### PR TITLE
[ALLUXIO-2560] Improving dependency and plugin management

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses https://alluxio.atlassian.net/browse/ALLUXIO-2560

Dependency versions are specified in the root module using dependency management.

Plugin versions are specified in the root module using plugin management.

All dependencies are sorted as follows:
1. dependencies are divided into external, internal, external test, and internal test groups
2. within each group dependencies are sorted lexicographically by group ID and artifact ID

The `maven-dependency-plugin` is used to remove unused declared dependencies and explicitly declare previously used but undeclared dependencies. This results in 10% reduction of assembly uber jar size and is now enforced through a build plugin.

Dependency exclusions are cleaned up and justified with a comment.

### CODE REVIEW

File: core/client/pom.xml [hunk 1]
```diff
@@ -160,82 +208,98 @@
...
-          <scope>provided</scope>
-        </dependency>
       </dependencies>
     </profile>
   </profiles>
 
 
   <build>
     <plugins>
+      <plugin>
```
Comment (apc999) [inline core/client/pom.xml]: could we reply on maven [Plugin_Management](https://maven.apache.org/pom.html#Plugin_Management) here to simply the per-pom setup work?

File: core/server/common/pom.xml [hunk 2]
```diff
@@ -33,63 +33,89 @@
...
+      <artifactId>commons-lang3</artifactId>
+    </dependency>
+    <dependency>
       <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-client</artifactId>
-      <!--
-          Newer versions of hadoop-client include the old javax.servlet:servlet-api jar.
-          This is excluded to prevent conflicts with the newer javax.servlet:javax.servlet-api jar.
-      -->
-      <exclusions>
```
Comment (apc999) [inline core/server/common/pom.xml]: are we fine with removing this exclusion?

File: keyvalue/common/pom.xml [hunk 3]
```diff
@@ -30,10 +30,42 @@
...
+
+    <!-- Internal dependencies -->
     <dependency>
       <groupId>org.alluxio</groupId>
       <artifactId>alluxio-core-common</artifactId>
       <version>${project.version}</version>
     </dependency>
   </dependencies>
+
+  <build>
```
Comment (apc999) [inline keyvalue/common/pom.xml]: vanilla use of dependnecy-plugin, I believe they could be simplied by plugin manager

**Comment by jsimsa** [comment]
Jenkins, test this please

**Pull request closed without merging**